File: searchInsert/searchInsert.py
def search_insert(arr: list[int], target: int) -> int:
    left = 0
    right = len(arr)

    while left < right:
        middle = left + (right - left) // 2

        if arr[middle] < target:
            left = middle + 1
        else:
            right = middle

    return left

File: searchInsert/searchInsert.ts
function serchInsert(nums: number[], target: number): number {
    let left = 0;
    let right = nums.length;
    while (left < right) {
        const middle = left + Math.floor((right - left) / 2);
        if (nums[middle] < target) {
            left = middle + 1;
        } else {
            right = middle;
        }
    }
    return left;
}